### PR TITLE
Documentation spring cleanup

### PR DESCRIPTION
**Pull request by tilsche**

### CODE REVIEW

**Comment by tilsche** [comment]
Got a bigger as I started cleaning up warnings while adding the migration guideline.

File: metricq/synchronous_source.py [hunk 1]
```diff
@@ -128,6 +151,19 @@ def declare_metrics(
         block: bool = True,
         timeout: float = 60,
     ) -> None:
+        """
+        Declare the metrics that are published by this source.
+        Returns immediately unless `block` is `True`.
+        Exceptions other than a timeout are not propagated to the caller, but logged instead.
+
+        Args:
+            metrics: a mapping from metric name to metadata
```
Comment (bmario) [inline metricq/synchronous_source.py:160]: The type of that parameter should be `Mapping[Metric, MetadataDict]`

File: metricq/synchronous_source.py [hunk 2]
```diff
@@ -80,6 +80,14 @@ def run(self, *args: Any, **kwargs: Any) -> None:
 
 
 class SynchronousSource:
+    """
+    This is a :term:`Source` that can be used in a synchronous context.
+    It spawns a new thread and runs an asynchronous :class:`Source` in it.
+    Therefore, it this class does not actually derive from :class:`Source`.
```
Comment (bmario) [inline metricq/synchronous_source.py:86]:
```suggestion
    Therefore, this class does not actually derive from :class:`Source`.
```

File: docs/howto/sink.rst [hunk 3]
```diff
@@ -148,3 +148,16 @@ and run it like so:
 ----
 
 .. literalinclude:: /../examples/metricq_sink.py
+
+Durable / persistent sinks
+--------------------------
+
+Most sinks are transient and not unique and do not have a configuration.
+To create persistent sink with a configuration, subclass :class:`metricq.DurableSink` instead of :class:`metricq.Sink`.
+Further, you need to implement a RPC handler for `config`.
```
Comment (bmario) [inline docs/howto/sink.rst:157]:
```suggestion
Further, you need to implement an RPC handler for `config`.
```

File: docs/api/sink.rst [hunk 4]
```diff
@@ -6,3 +6,6 @@ Sink
         connect,
         subscribe,
         on_data,
+
```
Comment (bmario) [inline docs/api/sink.rst:9]:
```suggestion

----
```

**Review by bmario — CHANGES_REQUESTED** [review]

**Pull request merged**